src/lib/staged_ledger/staged_ledger.mli

```diff
@@ -131,6 +133,10 @@ module Scan_state : sig
              Or_error.t )
     -> t
     -> unit Deferred.Or_error.t
+
+  val generate : Stable.Latest.t -> t
```

Comment:
These names aren't very clear to anyone who isn't actively reviewing this PR series. It's somewhat more annoying, but more explicit names like `write_all_proofs_to_disk` and `read_all_proofs_from_disk` makes it much clearer what's going on and what the consequences are.

Reply:
I'll go with `load_cached_proofs` and `cache_proofs` then